### PR TITLE
Pivotal ID #177423348: PMC integration test

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/177423348

### CODE REVIEW

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt [hunk 1]
```diff
@@ -49,28 +47,28 @@ internal class TsvChunkGenerator(private val parser: CsvParser = createParser())
...
-         * Create CSV parser function which allow to quote special characters or multiline values as "the \n value" but
-         * at the same time allows using them explicitly like in 'the "final" version'.
-         *
-         */
-        private fun createParseSettings() = CsvParserSettings().apply {
-            format.setLineSeparator("\n")
-            format.delimiter = TAB
-            format.comment = TSV_COMMENT
-            skipEmptyLines = false
+        private fun createParser(): CSVFormat {
```
Comment (jhoanmanuelms) [inline commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt:64]: Why are you removing these changes back? I think these are the fix for https://www.pivotaltracker.com/story/show/174642598 right?

File: scheduler/pmc-processor-task/gradle.properties [hunk 2]
```diff
@@ -1 +1 @@
-coverage=1
+coverage=0
```
Comment (jhoanmanuelms) [inline scheduler/pmc-processor-task/gradle.properties:1]: 😢

File: scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/config/WebConfig.kt [hunk 3]
```diff
@@ -13,9 +14,9 @@ import java.util.concurrent.TimeUnit
 class WebConfig {
 
     @Bean
-    fun pmcApi(): PmcApi {
+    fun pmcApi(properties: PmcImporterProperties): PmcApi {
         return Retrofit.Builder()
-            .baseUrl("http://www.ft-loading.europepmc.org")
+            .baseUrl(properties.pmcBaseUrl)
```
Comment (jhoanmanuelms) [inline scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/config/WebConfig.kt:19]: 👍

File: scheduler/pmc-processor-task/src/test/kotlin/ac/uk/ebi/pmc/Constants.kt [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+const val FILE1_NAME = "file1.txt"
+const val FILE2_NAME = "file2.txt"
+const val URL_FILE1_FILES_SERVER = "/files/getFileStream/PMC1234567?filename=file1.txt"
+const val URL_FILE2_FILES_SERVER = "/files/getFileStream/PMC1234567?filename=file2.txt"
+const val URL_FILE3_FILES_SERVER = "/files/getFileStream/PMC8901234?filename=file3.txt"
+internal const val SUB_ERROR_TEXT = "Submission\tS-456ERROR\tPublic\n\nStudy\n\nLinks"
+internal const val ACC_NO = "S-123SUCCESS"
+internal val SUB_ATTRIBUTE = Attribute("Title", "Submission title")
+
+val submissionBody = jsonObj {
```
Comment (jhoanmanuelms) [inline scheduler/pmc-processor-task/src/test/kotlin/ac/uk/ebi/pmc/Constants.kt:28]: Perhaps these chould be in a separated factory file?

File: scheduler/pmc-processor-task/src/test/kotlin/ac/uk/ebi/pmc/PmcSubmissionProcessorTest.kt [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+import org.testcontainers.containers.MongoDBContainer
+import org.testcontainers.utility.DockerImageName
+import java.io.File
+import java.net.HttpURLConnection.HTTP_BAD_REQUEST
+import java.net.HttpURLConnection.HTTP_OK
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ContextConfiguration(classes = [PersistenceConfig::class])
+@ExtendWith(TemporaryFolderExtension::class)
+internal class PmcSubmissionProcessorTest(private val tempFolder: TemporaryFolder) {
```
Comment (jhoanmanuelms) [inline scheduler/pmc-processor-task/src/test/kotlin/ac/uk/ebi/pmc/PmcSubmissionProcessorTest.kt:49]: 🤘

File: scheduler/pmc-processor-task/src/test/resources/file1.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+that is the content
```
Comment (jhoanmanuelms) [inline scheduler/pmc-processor-task/src/test/resources/file1.txt:1]: couldn't we just create these test files with the temporary folder extension?

**Review by jhoanmanuelms — COMMENTED** [review]
A couple of questions

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt (hunk 1 above)
Comment (Juan-EBI) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt:64]: https://github.com/uniVocity/univocity-parsers/issues/454

File: scheduler/pmc-processor-task/src/test/resources/file1.txt (hunk 6 above)
Comment (Juan-EBI) [reply scheduler/pmc-processor-task/src/test/resources/file1.txt:1]: yes no point to add this a resource

**Review by jhoanmanuelms — APPROVED** [review]
LGTM

**Pull request merged**